MinimedKit/GlucoseEvents/SensorTimestampGlucoseEvent.swift

```diff
@@ -27,20 +48,14 @@ public struct SensorTimestampGlucoseEvent: ReferenceTimestampedGlucoseEvent {
         
         rawData = availableData.subdata(in: 0..<length)
         timestamp = DateComponents(glucoseEventBytes: availableData.subdata(in: 1..<5))
-        
-        switch (d(3) >> 5 & 0b00000011) {
-        case 0x00:
-            timestampType = "last_rf"
-        case 0x01:
-            timestampType = "page_end"
-        case 0x02:
-            timestampType = "gap"
-        default:
-            timestampType = "unknown"
-        }
+        timestampType = SensorTimestampType.eventType(code: UInt8(d(3) >> 5) & 0b00000011)
```

Comment:
Sorry for just noticing this now, but "`d`" is useless here as you're undoing the unnecessary `Int` cast. Coupled with the initializer change: 

```swift
timestampType = SensorTimestampType(code: availableData[3] >> 5 & 0b11)
```

is much cleaner